apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -565,30 +596,70 @@ def parse(element_html, data):
         element, "grading-method", GRADING_METHOD_DEFAULT
     )
     correct_answers = data["correct_answers"][answer_name]
+    blocks = data["params"][answer_name]
 
     if grading_mode == "ranking":
         for answer in student_answer:
-            search = next(
+            matching_block = next(
                 (
                     item
                     for item in correct_answers
                     if item["inner_html"] == answer["inner_html"]
                 ),
                 None,
             )
-            answer["ranking"] = search["ranking"] if search is not None else None
-            answer["tag"] = search["tag"] if search is not None else None
+            answer["ranking"] = (
+                matching_block["ranking"] if matching_block is not None else None
+            )
+            answer["tag"] = (
+                matching_block["tag"] if matching_block is not None else None
+            )
+            if matching_block is None:
+                matching_block = next(
+                    (
+                        item
+                        for item in blocks
+                        if item["inner_html"] == answer["inner_html"]
+                    ),
+                    None,
+                )
+                answer["distractor_feedback"] = (
+                    matching_block["distractor_feedback"]
+                    if matching_block is not None
+                    else None
+                )
     elif grading_mode == "dag":
         for answer in student_answer:
-            search = next(
+            matching_block = next(
                 (
                     item
                     for item in correct_answers
                     if item["inner_html"] == answer["inner_html"]
                 ),
                 None,
             )
-            answer["tag"] = search["tag"] if search is not None else None
+            answer["tag"] = (
```

Comment:
Same comments as above; this setting logic is very self-similar and can probably be cleaned up using some helper functions.

Reply:
I combined some if blocks instead to get rid of the code duplication.